#### I was given a messy tab seperated data file that will need some cleaning up prior to analysis. First, I import the necessary libraries and load in the data.

#### The dataframe will not accept large paragrapsh of text, for now I have removed columns with lots of text and saved a new version of the data file to allow for transparency of thought process. 

In [1]:
%matplotlib inline
import pandas as pd
fname = "../data/Funded_program_attributes_mod2.txt"

In [30]:
df0 = pd.read_csv(fname, sep='\t', header=0)
df0.count() 

org       6368
name      6368
tbl1      6353
value1    6364
value2    6364
value3    3828
value4    3860
value5    3772
value6    3856
type      6368
perm1        1
perm2        1
perm3        1
perm4        0
perm5        0
perm6        0
perm7        0
perm8        0
perm9        0
dtype: int64

#### The largest column count is 6368, but I must have a dollar value, so later, I look to column 'value4' which is shown to have 3860 line items.

#### Now I view the dataframe.

In [33]:
df0

org                   name  tbl1      value1      value2  \
0              0            Non-Program   NaN         NaN         NaN   
1     1ACOL2GENF     Assess to Gen Fund  OTHR  20000101.0  99991231.0   
2     1ARTS2GENF   Arts Edu & Acc to GF  OTHR  20131101.0  99991231.0   
3     1BBND2GENF   Bancrft Bnd to GenFd  OTHR  20000101.0  99991231.0   
4     1BLSF2CAMP   BL Surch to Camp Fin  OTHR  20000101.0  99991231.0   
5     1BLSF2GENF   BL Surch to Gen Fund  OTHR  20000101.0  99991231.0   
6     1CABL2CAMP      Cable to Camp Fin  OTHR  20000101.0  99991231.0   
7     1CABL2GENF      Cable to Gen Fund  OTHR  20000101.0  99991231.0   
8     1CABL2HLTH        Cable to Health  OTHR  20000101.0  99991231.0   
9     1CAMP2CABL      Camp Fin to Cable  OTHR  20000101.0  99991231.0   
10    1CAMP2CAMP   Camp Fin to Camp Fin  OTHR  20000101.0  99991231.0   
11    1CAMP2CFLT    Camp Fin to CityFlt  OTHR  20000101.0  99991231.0   
12    1CAMP2DEVS    Camp Fin to Dev Svs  OTHR  20000101.0  99991231.0   
13    1CAMP2ECOM   Camp Fin to Emer Com  OTHR  20000101.0  99991231.0   
14    1CAMP2ENVR    Camp Fin to Environ  OTHR  20000101.0  99991231.0   
15    1CAMP2FACS    Camp Fin to Fac Svs  OTHR  20000101.0  99991231.0   
16    1CAMP2FGOB    Camp Fin to Fire GO  OTHR  20000101.0  99991231.0   
17    1CAMP2FPDR       Camp Fin to FPDR  OTHR  20000101.0  99991231.0   
18    1CAMP2GENF   Camp Fin to Gen Fund  OTHR  20000101.0  99991231.0   
19    1CAMP2GOLF       Camp Fin to Golf  OTHR  20000101.0  99991231.0   
20    1CAMP2HYDR   Camp Fin to Hydro Op  OTHR  20000101.0  99991231.0   
21    1CAMP2INSC   Camp Fin to Ins Clms  OTHR  20000101.0  99991231.0   
22    1CAMP2LIDC   Camp Fin to LID Cons  OTHR  20000101.0  99991231.0   
23    1CAMP2PCON   Camp Fin to Park Con  OTHR  20000101.0  99991231.0   
24    1CAMP2PDOP     Camp Fin to P&D Op  OTHR  20000101.0  99991231.0   
25    1CAMP2PFAC   Camp Fin to Park Fac  OTHR  20000101.0  99991231.0   
26    1CAMP2PIRF        Camp Fin to PIR  OTHR  20000101.0  99991231.0   
27    1CAMP2PLSR   Camp Fin to Pol Spec  OTHR  20000101.0  99991231.0   
28    1CAMP2SEWR   Camp Fin to Sewer Op  OTHR  20000101.0  99991231.0   
29    1CAMP2SFAC   Camp Fin to Spec Fac  OTHR  20000101.0  99991231.0   
...          ...                    ...   ...         ...         ...   
6338      WMTR15    MTR Pur & Rep 14-15  CIPJ  20000101.0  99991231.0   
6339      WMTR16   FY 2015-16 METER Pur  CIPJ  20140701.0  99991231.0   
6340      WMTR17         FY 16-17 Meter  CIPJ  20160501.0  99991231.0   
6341      WP0000  Water Permit Projects  CIPJ   1012000.0  12319999.0   
6342      WPLN17      FY 16-17 Cap Plan  CIPJ  20160501.0  99991231.0   
6343      WRTU01       RTU Replacements  CIPJ  20000101.0  99991231.0   
6344      WRTU13   RTU Replace FYE 2013  CIPJ  20000101.0  99991231.0   
6345      WRTU14   RTU Replace FYE 2014  CIPJ  20000101.0  99991231.0   
6346      WRTU15       RTUs for FY14-15  CIPJ  20140630.0  99991231.0   
6347      WRTU16    RTU Replacements FY  CIPJ  20140701.0  99991231.0   
6348      WRTU17   FY 16-17 RTU Replace  CIPJ  20160501.0  99991231.0   
6349      WSVC01   Svc.Ord New 5/8 to 1  CIPJ  20000101.0  99991231.0   
6350      WSVC02   Svc Ord Actvte 5/8-1  CIPJ  20000101.0  99991231.0   
6351      WSVC03    Svcs Renewal 2"&Bel  CIPJ  20000101.0  99991231.0   
6352      WSVC04   Svcs Renewal 3"& Abv  CIPJ  20000101.0  99991231.0   
6353      WSVC05   Srvc/Fire Srvc 1.5-2  CIPJ  20000101.0  99991231.0   
6354      WSVC06      Srvc/FireSrvc 3in  CIPJ  20000101.0  99991231.0   
6355      WSVC07    Srvc 2 w/FireSrvc 3  CIPJ  20000101.0  99991231.0   
6356      WSVC08    Srvc w/FireSrvc 3in  CIPJ  20000101.0  99991231.0   
6357      WSVC13    All SVC for FY12-13  CIPJ  20000101.0  99991231.0   
6358      WSVC14    All SVC for FY13-14  CIPJ  20000101.0  99991231.0   
6359      WSVC15    All SVC for FY14-15  CIPJ  20000101.0  99991231.0   
6360      WSVC16    FY 2015-16 Services  CIPJ  20140701.0  99991231.0  

#### There are too many columns, many with only "NaN" meaning not a number, so to make the dataframe more manageable for now I will only select all data in a choice few columns.

In [36]:
# only have headers in dataframe you want
df1 = df0.loc[:,['org','name','tbl1','value1','value4']]
df1.head(20)

org                  name  tbl1      value1 value4
0            0           Non-Program   NaN         NaN    NaN
1   1ACOL2GENF    Assess to Gen Fund  OTHR  20000101.0    NaN
2   1ARTS2GENF  Arts Edu & Acc to GF  OTHR  20131101.0    NaN
3   1BBND2GENF  Bancrft Bnd to GenFd  OTHR  20000101.0    NaN
4   1BLSF2CAMP  BL Surch to Camp Fin  OTHR  20000101.0    NaN
5   1BLSF2GENF  BL Surch to Gen Fund  OTHR  20000101.0    NaN
6   1CABL2CAMP     Cable to Camp Fin  OTHR  20000101.0    NaN
7   1CABL2GENF     Cable to Gen Fund  OTHR  20000101.0    NaN
8   1CABL2HLTH       Cable to Health  OTHR  20000101.0    NaN
9   1CAMP2CABL     Camp Fin to Cable  OTHR  20000101.0    NaN
10  1CAMP2CAMP  Camp Fin to Camp Fin  OTHR  20000101.0    NaN
11  1CAMP2CFLT   Camp Fin to CityFlt  OTHR  20000101.0    NaN
12  1CAMP2DEVS   Camp Fin to Dev Svs  OTHR  20000101.0    NaN
13  1CAMP2ECOM  Camp Fin to Emer Com  OTHR  20000101.0    NaN
14  1CAMP2ENVR   Camp Fin to Environ  OTHR  20000101.0    NaN
15  1CAMP2FACS   Camp Fin to Fac Svs  OTHR  20000101.0    NaN
16  1CAMP2FGOB   Camp Fin to Fire GO  OTHR  20000101.0    NaN
17  1CAMP2FPDR      Camp Fin to FPDR  OTHR  20000101.0    NaN
18  1CAMP2GENF  Camp Fin to Gen Fund  OTHR  20000101.0    NaN
19  1CAMP2GOLF      Camp Fin to Golf  OTHR  20000101.0    NaN